vault/mount.go

```diff
@@ -479,6 +575,85 @@ func (c *Core) newLogicalBackend(t string, view logical.Storage, conf map[string
 	return b, nil
 }
 
+// mountEntrySysView creates a logical.SystemView from global and
+// mount-specific entries
+func (c *Core) mountEntrySysView(me *MountEntry) (logical.SystemView, error) {
+	if me == nil {
+		return nil, fmt.Errorf("[ERR] core: nil MountEntry when generating SystemView")
+	}
+
+	sysView := dynamicSystemView{
+		core: c,
+		path: me.Path,
+	}
+
+	return sysView, nil
+}
+
+// sysViewByPath is a simple helper for MountEntrySysView
+func (c *Core) sysViewByPath(path string) (logical.SystemView, error) {
+	// Ensure we end the path in a slash
+	if !strings.HasSuffix(path, "/") {
+		path += "/"
+	}
+
+	me, err := c.mountEntryByPath(path)
+	if err != nil {
+		return nil, err
+	}
+	return c.mountEntrySysView(me)
+}
+
+// mountEntryByPath searches across all tables to find the MountEntry
+func (c *Core) mountEntryByPath(path string) (*MountEntry, error) {
```

Comment:
I would just add a `MatchingSystemView` to Router like `MatchingView` already exists and do the lookup there. Much more efficient since its already using radix trees to index by path.